src/metrics/metric/id/m3/id.go

```diff
@@ -79,18 +79,9 @@ func NewRollupID(name []byte, tagPairs []id.TagPair) []byte {
 }
 
 // IsRollupID determines whether an id is a rollup id.
-// Caller may optionally pass in a sorted tag iterator
-// pool for efficiency reasons.
 // nolint: unparam
-func IsRollupID(name []byte, tags []byte, iterPool id.SortedTagIteratorPool) bool {
-	var iter id.SortedTagIterator
-	if iterPool == nil {
-		iter = NewSortedTagIterator(tags)
-	} else {
-		iter = iterPool.Get()
-		iter.Reset(tags)
-	}
-	defer iter.Close()
+func IsRollupID(name []byte, tags []byte) bool {
+	iter := newPooledSortedTagIterator(tags, nil)
```

Comment:
the fact that you're not calling closed makes me think you don't want a pooled iterator at all?

Reply:
100% correct yeah--I'm avoiding the pooled code path here (passing nil for pool)